packages/flutter/lib/src/cupertino/dialog.dart

```diff
@@ -68,11 +69,22 @@ class CupertinoDialog extends StatelessWidget {
   @override
   Widget build(BuildContext context) {
     return new Center(
-      child: new Container(
-        margin: const EdgeInsets.all(10.0),
-        width: _kCupertinoDialogWidth,
-        decoration: _kCupertinoDialogDecoration,
-        child: child,
+      child: new ClipRRect(
```

Comment:
You might not need this ClipRRect.  Instead, try applying the shape to the decorations.  I think the blur effect uses the alpha channel to determine where to apply.